include/RAJA/util/KokkosPluginLoader.hpp

```diff
@@ -0,0 +1,56 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_Kokkos_Plugin_Loader_HPP
+#define RAJA_Kokkos_Plugin_Loader_HPP
+
+#include <memory>
+#include <vector>
+
+#include "RAJA/util/PluginOptions.hpp"
+#include "RAJA/util/PluginStrategy.hpp"
+
+namespace RAJA {
+namespace util {
+
+  class KokkosPluginLoader : public ::RAJA::util::PluginStrategy
+  {
+    using Parent = ::RAJA::util::PluginStrategy;
+    typedef void (*init_function)(const int, const uint64_t, const uint32_t, void*);
+    typedef void (*pre_function)(const char*, const uint32_t, uint64_t*);
+    typedef void (*post_function)(uint64_t);
+    typedef void (*finalize_function)();
+
+  public:
```

Comment:
Please move the public section of the class to the top. It's easier for users to see the interface in the docs that way. Most users shouldn't need to look at the implementation so it shouldn't be the first thing they see.